backoffice/src/components/TableWrapper/index.tsx

```diff
@@ -9,27 +9,43 @@ import {
   TablePagination,
   Select,
   MenuItem,
-  CardHeader
+  CardHeader,
+  Button
 } from '@mui/material';
-
+import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
 import BulkActions from './BulkActions';
-import { StatusEnum } from 'src/models/status';
+import { StatusEnum, StatusOptions } from 'src/models/status';
+import { addItem, deleteItem, getItems, updateItem } from 'src/services/commons.service';
+import { useModal } from 'src/hooks/useModal';
 
 interface TableWrapperProps {
   className?: string;
-  items: any;
+  defaultItems: any;
+  url: string,
   title?: string,
-  statusOptions?: StatusOption[],
-  children: ReactNode
+  statusOptions?: StatusOptions,
+  children: ReactNode,
+  addButton?: boolean
 }
 
-interface Filters {
-  status?: StatusEnum | 'all';
+export interface TableProps<T> {
+  items: T[], 
+  selectedItems: any,
+  handleSelectAllItems: (event: ChangeEvent<HTMLInputElement>) => void, 
+  handleSelectOneItem: (event: ChangeEvent<HTMLInputElement>, itemId: string) => void,
+  selectedSomeItems: any,
+  selectedAllItems: any,
+  handleDeleteItem: (id: string) => any,
+  handleUpdateItem: (id: string, data: object) => any,
+  handleAddItem: (data: object) => any,
```

Comment:
tu peux remplacer les any de ces functions par Promise<Type> ou Promise<void> suivant ce qui est renvoyé

```suggestion
  handleDeleteItem: (id: string) => Promise<Type>,
  handleUpdateItem: (id: string, data: object) => Promise<Type>,
  handleAddItem: (data: object) => Promise<Type>,
```

Reply:
merci !